make/mt-os/mt-esp32.mk

```diff
@@ -5,13 +5,76 @@
 
 include make/mt-os/mt-os-common.mk
 
-#$(IDF_PATH) This is exported in the shell as as IDF requirement
-XI_ESP_IDF_SDK_PATH ?= $(IDF_PATH)
-
 # The path to CC and AR must be in the environment $PATH as an IDF requirement
 CC = xtensa-esp32-elf-gcc
 AR = xtensa-esp32-elf-ar
 
+XI_ESP32_PREREQUISITE_DOWNLOAD_PATH := $(HOME)/Downloads/esp32
+
+################################
+# auto-provide ESP32 SDK #######
+################################
+ifeq (,$(wildcard $(IDF_PATH)))
+    $(info ESP32 SDK is NOT available, using auto-downloaded ESP32 SDK from $(XI_ESP32_PREREQUISITE_DOWNLOAD_PATH))
+
+    XI_ESP_IDF_SDK_PATH := $(XI_ESP32_PREREQUISITE_DOWNLOAD_PATH)/esp-idf
```

Comment:
Can the makefile export `XI_ESP_IDF_SDK_PATH` into the user's shell environment? They'll need it when they try to build the application. If exporting it is not an issue, at least we should print a message along the lines of:

```
NOTE: The environment variable IDF_PATH was not found in your system, so we've downloaded the SDK for you. You can find it in folder $(XI_ESP_IDF_SDK_PATH).
You'll need to export it to your environment using this command before compiling your application:
`export IDF_PATH=$(XI_ESP_IDF_SDK_PATH)`
```

Same thing the toolchain in the environment's `PATH`

Reply:
right, good point, thinking...